### DISCUSSION TITLE
Cloud functions

### DISCUSSION THREAD

**Discussion by charbs-io** [pockethost/pockethost#249] [Q&A]
Hi, now that [this](https://github.com/pocketbase/pocketbase/discussions/2825) seems to have hit a wall, i'm interested in revisitng PH's cloud functions.

Are they still in beta? I can't find much documentation on them (Looked [here](https://pockethost.gitbook.io/manual/daily-usage/cloud) and in these discussion threads to no avail).

Is anyone able to point me in the right direction please?

**Comment by benallfree** [comment]
Also being discussed in #247. #27  and #76 also have some discussion around this.

Right now, you can upload `worker/index.ts` and PocketHost will launch it along with your instance. Here is an example https://github.com/benallfree/pockethost/blob/master/packages/examples/worker/src/index.ts

```ts
import { init } from '../../../deno-worker/index.ts'

import PocketBase from 'https://cdn.jsdelivr.net/npm/pocketbase'

const { client, adminAuthWithPassword } = init(PocketBase)

try {
  await adminAuthWithPassword()
  setInterval(() => {
    console.log(`ping`)
  }, 1000)
} catch (e) {
  console.error(`caught an error`, e, JSON.stringify(e))
}
```

In order to authenticate as admin, you need to add some [secrets](https://pockethost.gitbook.io/manual/daily-usage/index#secrets) to your instance. As you can see, it's still a work in progress. We haven't yet released the `pockethost-worker` package for Deno, but you can copy it from `packages/deno-worker/src/index.ts` and try it for yourself.
